flocker/provision/test/test_ssh_keys.py

```diff
@@ -0,0 +1,86 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for :module:`flocker.provision._ssh._keys`.
+"""
+
+import os
+
+from twisted.internet import reactor
+from twisted.python.filepath import FilePath
+from twisted.trial.unittest import TestCase
+
+from flocker.testtools.ssh import create_ssh_agent, generate_ssh_key
+
+from .._ssh._keys import ensure_agent_has_ssh_key, KeyNotFound, AgentNotFound
+
+
+class EnsureKeyTests(TestCase):
```

Comment:
:question: All tests set `key_file = FilePath(self.mktemp())` so this can be in setUp.